lib/compress/zstd_opt.c

```diff
@@ -818,6 +968,17 @@ ZSTD_compressBlock_opt_generic(ZSTD_matchState_t* ms,
     ZSTD_match_t* const matches = optStatePtr->matchTable;
     ZSTD_optimal_t lastSequence;
 
+    U32 ldmStartPosInBlock = 0;
+    U32 ldmEndPosInBlock = 0;
+    U32 ldmOffset = 0;
```

Comment:
These 3 variable are all related and modified together. And in fact the optimal parser doesn't care about these variables at all, it only matters for the LDM helper functions.

These should be grouped together into a struct. I'd probably also include the `&ms->ldmSeqStore` in this struct.

```c
typedef struct {
  rawSeqStore* seqStore;
  U32 startPosInBlock;
  U32 endPosInBlock;
  U32 offset;
} ZSTD_optLdm_t;
```

Then you might want to rename all the functions to be `ZSTD_optLdm_blah(ZSTD_optLdm_t* optLdm, ...)` to be clear it is operating on this struct.